doc/embedded/supported_boards.md

```diff
@@ -0,0 +1,73 @@
+## List of supported architectures:
+
+The following table contains a list of different architectures that you can
+compile or cross compile mlpack to, this page assumes that you have a
+functional operating system (e.g., Linux, MacOS, Windows) with compatible
+ABI on both the host and the target. In addition, your objective is to create
+an embedded ABI (eabi). For now, this guide does not support none-eabi or
+bare-metal C/C++. The following table refers only to possible values of 
+`BOARD_NAME` CMake flag that needs to be specified before the compilation
+process starts.
+
+please check its architecture of your board before specifying the parameter, if
+your architecture is not part of the board please refer to the closest
+architecture with similar word size, or adapt the parameters and feel free to
+open a pull request.
+
+If your device has a fully functional package manager such as debian, and you
+have enough RAM on your embedded device (e.g., RAM > 4GB), then you will
+probably be able to compile directly on the device. However, if these are not
+available then you will need to follow these instruction to do cross
+compilation.
+
+Setting up the cross compilation toolchain is basically easy. We usually use
+bootlin [toolchains](https://toolchains.bootlin.com/) but this is not
+obligatory. If you have a specific architecture, then you need you need to
+identify two parameters that you need fill them in the first one is the
+`TOOLCHAIN_PREFIX` and the second one is the `DCMAKE_SYSROOT`. For more
+detailed example please check this [tutorial](crosscompile_armv7.md)
+
+Here is the generic CMake command, please check the table to modify it
+accoding to each architecture and the version of your compiler set, also
+do not forget to change `/path/to/bootlin/toolchain` by the real path:
+
+```sh
+cmake \
+    -DBUILD_TESTS=ON \
+    -DBOARD_NAME="ARM11" \
+    -DCMAKE_CROSSCOMPILE=ON \
+    -DCMAKE_TOOLCHAIN_FILE=../board/crosscompile-toolchain.cmake \
+    -DTOOLCHAIN_PREFIX=(Check the following table) \
+    -DCMAKE_SYSROOT=(Check the following table) \
+    ../
+```
+
+| Architecture\_NAME | link to crosscompiler | CMake command |  
```

Comment:
With the CMake command in the table itself, it actually gets way too wide to read:

http://ci.mlpack.org/job/pull-request%20documentation%20build%20and%20test/911/Built_20documentation/embedded/supported_boards.html

I'm trying to think about how to fix this... I can see a couple possibilities but I'm not sure if I have an opinion which is best:

 * Change the CMake command column to a link to an anchor (like `[caption](#arm11)`) and then have an `ARM11` section that just has the command.  A little similar to how it was before, it just adds a link to the table.
 * Omit the CMake command column entirely, so the user has to find the toolchain directory and fill it in themselves.  Maybe that is actually more "futureproof" as bootlin releases new versions too.

Also it would be really nice to add a column with a handful of devices that have the architecture; what do you think?  I can help dig around to find example devices.  The thought is just that a user comes to this page knowing the product they have (e.g. "I have a BAE RAD750" although I'm not sure that exact case will happen much :smile:) and then they need to map that product or board to the actual name of the architecture that they should use.

Reply:
The first one seems better to me, providing the CMake command can really facilitate a lot of headaches, and then it is to the use to find which one.
And yes, I agree, you can choose the first one but not necessarily, you also have another compiler set, which is why I am trying not to mention anything. The user should be able to select the toolchain that they are looking for telling them which one to choose is beyond our objectives